compiler/rustc_hir_typeck/src/demand.rs

```diff
@@ -1417,6 +1417,21 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         }
                     }
 
+                    if let Some(hir::Node::Expr(hir::Expr {
+                        kind: hir::ExprKind::Binary(_, left, right),
+                        ..
+                    })) = self.tcx.hir().find_parent(expr.hir_id) && mutability.is_mut() {
+                        let deref_expr = if expr.hir_id == left.hir_id { right } else { left };
+                        let sugg = vec![(deref_expr.span.shrink_to_lo(), "*".to_string())];
+                        return Some((
+                            sugg,
+                            format!("consider dereference here"),
```

Comment:
```suggestion
                            format!("consider dereferencing here"),
```

Further, I think it'd make sense to use a translatable diagnostic here, so less needs to be migrated later on.

Reply:
> I think it'd make sense to use a translatable diagnostic here

The return type of `suggest_deref_or_ref` is a tuple instead of `Diagnostic`, so I think it would be better to change it in another PR